rebasehelper/application.py

```diff
@@ -193,6 +195,45 @@ def _prepare_spec_objects(self):
                 # parse spec again with sources downloaded to properly expand %prep section
                 spec_file.update()
 
+    def _sanitize_sources(self) -> None:
+        """Renames local sources whose name changed after version bump.
+
+        For example if the specfile contains a Patch such as %{name}-%{version}.patch,
+        the name changes after changing the version and the rebase would fail due to
+        a missing patch whilst building SRPM.
+
+        This method tries to correct such cases and rename the local file to match
+        the new name. Modifies the rebase_spec_file object to contain the correct
+        paths.
+        """
+        for source, index, source_type in self.rebase_spec_file.spc.sources:
+            if urllib.parse.urlparse(source).scheme or source_type == 0:
+                continue
+            if os.path.exists(os.path.join(self.execution_dir, source)):
+                continue
+            # Find matching source in the old spec
+            sources = [n for n, i, t in self.spec_file.spc.sources if i == index and t == source_type]
+            if not sources:
+                logger.error('Failed to find the source corresponding to %s in old version spec', source)
+                continue
+            source_old = sources[0]
+
+            # rename the source
+            old_source_path = os.path.join(self.rebased_sources_dir, source_old)
+            new_source_path = os.path.join(self.rebased_sources_dir, source)
+            logger.debug('Renaming %s to %s', old_source_path, new_source_path)
+            try:
+                os.rename(old_source_path, new_source_path)
+            except OSError:
+                logger.error('Failed to rename %s to %s while sanitizing sources', old_source_path, new_source_path)
+
+            # prepend the Source/Path with rebased-sources directory in the specfile
+            to_prepend = os.path.relpath(self.rebased_sources_dir, self.execution_dir)
+            tag = ('Patch{}' if source_type == 2 else 'Source{}').format(index)
```

Comment:
I would prefer this: `tag = '{0}{1}'.format('Patch' if source_type == 2 else 'Source', index)`